### ISSUE TITLE
cannot launch node of type RCT ROS TOOL

### ISSUE THREAD

**Issue by MARHEM602** [Jmeyer1292/robot_cal_tools#105]
Hello I'm doing a robotics project, and I'm trying to use your package for the calibration of my fanuc robot. Everything works except the part with the nodes, I still have the same problem (cannot launch node of type...). I thank you in advance

**Comment by marip8** [comment]
Can you provide more information about which launch file is causing the issue?

**Comment by Theopetitjean** [comment]
Hello ! 
Sorry for the late answer, i'll give you some more information about what me and my colleague are doing RN. 

We are under a Ubuntu 18.04, with ROS Melodic. We are trying to use your package to do the calibration work for a Fanucs Cr7iaL robotic manipulator. We have 2 camera avaliable : Kinect camera and a webcam C920 logitech. 

Since we are both kinda new to the usage of such kind of package, we acheive to use each part in separate way. The fanus Robot with the moveit package is ready to use with the provided ROS-INDUSTRIAL package that deal with FANUC arm. Same for te launch of the camera and etc. 
But when it come to use the package we face some issue, probably due to a miss from our side. 

So here is some of the prblm we face while trying to use the package : 

1) We cannot use classical CatKinMake command so we have to use CatKinMake_isolated
![image](https://user-images.githubusercontent.com/60598493/219317562-22e6ce0e-fc16-4ca1-a108-6f1b0b3ad702.png)

2) Then once the build is done we are facing some probleme regarding to the launch / nodes localisation such as : 
`ERROR: cannot launch node of type [rct_examples/moving_cam_extr_cal_ex]: Cannot locate node of type [moving_cam_extr_cal_ex] in package [rct_examples]. Make sure file exists in package path and permission is set to executable (chmod +x)`

3) We tried several way to fix the previous err. Using chmod+and make sure that everything is executable. But the fact is the file required don't exist in the repo. when i go in Rct_exemple/moving_cam_extr_cal_ex it simply dosen't exist. 

So here are my questions, and i hope i provided enought details on what we are actually facing some trouble. 

- Do you think that the isolated make could be a problem regarding to the application we are trying to put in place ? 
- Do we launch the right file or did we miss something ? We actually use the command : 
`roslaunch roslaunch rct_examples camera_on_wrist_example.launch camera_file:=<PATH_TO_YOUR_CAMERA_YAML> target_file:=<PATH_TO_YOUR_TARGET_DEF> data_path:=<PATH_TO_YOUR_DATA_INDEX_YAML>` with the specified path and so on. Shall we execute some other command or modify some stuff ? 

Anyway if you read this thanks for the time you give us. We probably did some very basic mistake, we are beginner and the spirit is not that easy to get. 

Also do not hesitate to point out if we did something really wrong ! ;)

**Comment by marip8** [comment]
I usually build this my ROS packages using [`catkin_tools`](https://catkin-tools.readthedocs.io/en/latest/) (i.e., `catkin build ...`, not `catkin_make` or `catkin_make_isolated`) and it was able to build all of the packages correctly. That is also what the CI build for this project uses. I was able to recreate your specific build problem by using `catkin_make` though. It seems like the issue for `catkin_make` is that some of the packages in this repository (`rct_common`, `rct_optimizations`, and `rct_image_tools`) are "pure CMake" packages that are independent of ROS/catkin, and `catkin_make` does not understand how to build them. Invoking `catkin_make_isolated` does successfully build all of the packages for me, and it seems that works for you as well. However, when you build with `catkin_make_isolated`, the build artifacts are installed to folders with the suffix `_isolated` (e.g., `build_isolated`, `devel_isolated`, `install_isolated`) instead of the "regular" build artifact folders. You'll need to source the `setup.bash` file from one of these folders (usually `devel_isolated`) instead, and then things should run correctly.

TLDR, either:
- build using `catkin_tools`, source and run as usual (recommended; `catkin_tools` is a much more user-friendly build tool than `catkin_make`)
- build using `catkin_make_isolated` and source the `devel_isolated/setup.bash` file instead of `devel/setup.bash`
